Mathlib/FieldTheory/IsAlgClosed/Basic.lean

```diff
@@ -265,6 +265,15 @@ noncomputable instance (priority := 100) perfectRing (p : ℕ) [Fact p.Prime] [C
   PerfectRing.ofSurjective k p fun _ => IsAlgClosed.exists_pow_nat_eq _ <| NeZero.pos p
 #align is_alg_closed.perfect_ring IsAlgClosed.perfectRing
 
+noncomputable instance (priority := 100) perfectField [IsAlgClosed k] : PerfectField k := by
+  obtain ⟨p, hchar⟩ := CharP.exists k
+  by_cases hp : p = 0
+  · haveI := (CharP.charP_zero_iff_charZero k).1 (hp ▸ hchar)
+    exact PerfectField.ofCharZero k
+  haveI := NeZero.mk hp
+  haveI := CharP.char_is_prime_of_pos k p
```

Comment:
I think there is a `charZero_or_charP` but I can't find it. Maybe there is no such result yet.

Reply:
Oh, it's [CharP.char_is_prime_or_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/CharP/Basic.html#CharP.char_is_prime_or_zero).

Reply:
It still doesn't give you a Fact instance directly, and you need to combine it with `CharP.exists`. I think it's worthwhile to add the lemma `CharZero k ∨ ∃ p : ℕ, Fact p.Prime ∧ CharP k p` because you get all the instances in place with `obtain h | ⟨p, _, _⟩ :=` in one stretch.

Reply:
Added `CharP.exists'` in #9573.